omegaconf/nodes.py

```diff
@@ -163,6 +163,126 @@ def __deepcopy__(self, memo: Dict[int, Any] = {}) -> "AnyNode":
         return res
 
 
+class UnionNode(ValueNode):
+    def __init__(
+        self,
+        value: Any = None,
+        key: Any = None,
+        parent: Optional[Container] = None,
+        is_optional: bool = True,
+        ref_type: Any = Union[Any],
+        element_types: List[Any] = None,  # type: ignore
```

Comment:
`[]`was the default before but, mypy didn't like that because it was mutable and I changed it.

Reply:
Right, [] is indeed discouraged, I can think of two "clean" options:
1. (The most common solution in such a situation) `self.element_types = element_types if element_types is not None else []`
2. (Make it a mandatory keyword arg, which may be better here because a `Union` with empty element_types makes no sense anyway):
```python
    def __init__(
        self,
        value: Any = None,
        key: Any = None,
        parent: Optional[Container] = None,
        is_optional: bool = True,
        ref_type: Any = Union[Any],
        *,
        element_types: List[Any],
    ):
```

Reply:
I like the second option.